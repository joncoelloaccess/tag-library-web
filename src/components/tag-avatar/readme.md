# tag-avatar

















<!-- Auto Generated Below -->


## Properties

#### placeholder

string

placeholder text when there is no image


#### size

string

size of the avatar: default 45px small 36px


#### src

string

source of the avatar image


## Attributes

#### placeholder

string

placeholder text when there is no image


#### size

string

size of the avatar: default 45px small 36px


#### src

string

source of the avatar image



----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
