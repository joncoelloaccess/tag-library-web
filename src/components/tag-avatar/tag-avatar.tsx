import { Component, Prop } from "@stencil/core";

type AvatarProps = {
  class: string;
  style?: {
    backgroundImage?: string;
  };
};

@Component({
  tag: "tag-avatar",
  styleUrl: "tag-avatar.scss",
  shadow: true
})
export class TagAvatar {
  /** placeholder text when there is no image */
  @Prop() placeholder: string;
  /** size of the avatar: default 45px small 36px */
  @Prop() size: string;
  /** source of the avatar image */
  @Prop() src: string;
  render() {
    const { src, placeholder, size } = this;
    const placeholderText = src ? "\u00a0" : placeholder;
    const avatarProps: AvatarProps = {
      class: `avatar ${size}`
    };
    if (src) {
      avatarProps.style.backgroundImage = `url(${src})`;
    }
    return <div {...avatarProps}>{placeholderText}</div>;
  }
}
