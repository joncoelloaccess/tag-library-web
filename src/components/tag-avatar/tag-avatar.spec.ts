import { render, flush } from "@stencil/core/testing";
import { TagAvatar } from "./tag-avatar";

describe("tag-avatar", () => {
  it("should build", () => {
    expect(new TagAvatar()).toBeTruthy();
  });

  describe("rendering", () => {
    let element;
    beforeEach(async () => {
      element = await render({
        components: [TagAvatar],
        html: "<tag-avatar></tag-avatar>"
      });
      console.log("@");
    });
    it("should render", async () => {
      await flush(element);
      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
