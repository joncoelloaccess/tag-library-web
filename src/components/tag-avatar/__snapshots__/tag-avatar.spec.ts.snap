// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag-avatar rendering should render 1`] = `"<tag-avatar data-tag-avatar-host=\\"\\" class=\\"hydrated\\"><!----><div class=\\"avatar undefined\\" data-tag-avatar=\\"\\"></div></tag-avatar>"`;
